### ISSUE TITLE
extruder problems

### ISSUE THREAD

**Issue by battistipaolo** [Ultimaker/Cura#7794]
Hi,
I have an over-extrusion problem only on the supports, so big that the extruder motor skips steps, the body of the piece is molded regularly.
ultimaker cura 4.6.1

[E.zip](https://github.com/Ultimaker/Cura/files/4656352/E.zip)
regards
paolo

**Comment by 64Florian** [comment]
Hello @battistipaolo,
I have try to play your GCode in Cura and I noticed the nozzle made 2 pass on support ...

Could you made a "save as" in Cura with you current settings ans opened STL file, zip this file and add in this ticket please, it could be easier to see if it's a setting issue ...

Just to be sure,
Did you use a older version of Cura on you computer and made an update to 4.6.1 or is it a fresh new version ? 
If you made an update, or have already an older version on your computer, could you go in Cura -> Help -> show configuration folder 
and zip the 4.6 folder to add in the ticket too please

Thank a lot

**Comment by 64Florian** [comment]
Today I see a similar issue when I use the "modify settings for overlapping" and use tree tree support settings ...
I'm pretty sure it's not your case .. but I post just to be sure ...
In my point of view in this case it's not needed to use the "modify settings for overlapping" setting

So, I manage to have a good result (in the preview) with following tree settings: 
![Capture d’écran 2020-05-21 à 11 19 25](https://user-images.githubusercontent.com/34726380/82544282-3318a500-9b55-11ea-979d-c07c45128a49.png)
![Capture d’écran 2020-05-21 à 11 24 16](https://user-images.githubusercontent.com/34726380/82544623-b0dcb080-9b55-11ea-8e11-834ce3fc2043.png)

**Comment by battistipaolo** [comment]
Hi ,
I try to send you zipped 4.6.1, it's an update to from 4.6.0.
I try to send you zipped 4.6.1, it's an update to from 4.6.0. 
That's 224 megabits how can I send it to you?
thank you for your interest.regards 
paolo battisti
    Il giovedì 21 maggio 2020, 04:18:21 CEST, TFlorian <notifications@github.com> ha scritto:  
 
 


Hello @battistipaolo,
I have try to play your GCode in Cura and I noticed the nozzle made 2 pass on support ...

Could you made a "save as" in Cura with you current settings ans opened STL file, zip this file and add in this ticket please, it could be easier to see if it's a setting issue ...

Just to be sure,
Did you use a older version of Cura on you computer and made an update to 4.6.1 or is it a fresh new version ?
If you made an update, or have already an older version on your computer, could you go in Cura -> Help -> show configuration folder
and zip the 4.6 folder to add in the ticket too please

Thank a lot

**Comment by battistipaolo** [comment]
this is the current setting
 paolo battisti
    Il giovedì 21 maggio 2020, 04:18:21 CEST, TFlorian <notifications@github.com> ha scritto:  
 
 


Hello @battistipaolo,
I have try to play your GCode in Cura and I noticed the nozzle made 2 pass on support ...

Could you made a "save as" in Cura with you current settings ans opened STL file, zip this file and add in this ticket please, it could be easier to see if it's a setting issue ...

Just to be sure,
Did you use a older version of Cura on you computer and made an update to 4.6.1 or is it a fresh new version ?
If you made an update, or have already an older version on your computer, could you go in Cura -> Help -> show configuration folder
and zip the 4.6 folder to add in the ticket too please

Thank a lot

**Comment by 64Florian** [comment]
Hello @battistipaolo,
Are you really sure it's the configuration folder only (_not all the application_) ? 
![Capture d’écran 2020-05-21 à 15 29 02](https://user-images.githubusercontent.com/34726380/82564108-7fc1a780-9b78-11ea-9362-f903f5ed7011.png)

This folder should be more or less 10 MB not more (and even less when zipped)

**Comment by 64Florian** [comment]
If you use windows: 
![curaWindows](https://user-images.githubusercontent.com/34726380/82564976-f4e1ac80-9b79-11ea-8336-878a4173d779.png)

**Comment by battistipaolo** [comment]
Hi, 

no no it's the whole folder "Ultimaker Cura 4.6". which folder do you want?
paolo

    Il giovedì 21 maggio 2020, 15:36:30 CEST, TFlorian <notifications@github.com> ha scritto:  
 
 


Hello @battistipaolo,
Are you really sure it's the configuration folder only (not all the application) ?


This folder should be more or less 10 MB not more (and even less when zipped)

**Comment by battistipaolo** [comment]
do you want this folder?
paolo

    Il giovedì 21 maggio 2020, 15:36:30 CEST, TFlorian <notifications@github.com> ha scritto:  
 
 


Hello @battistipaolo,
Are you really sure it's the configuration folder only (not all the application) ?


This folder should be more or less 10 MB not more (and even less when zipped)

**Comment by battistipaolo** [comment]
Hi,
I thought it was a problem of an extruder motor defect, but I changed the motor and the whole extruder line (nozzle, tube, head) but nothing changed, at that point I decided to consult you, also because with "kisslicer64" it doesn't .
paolo

    Il giovedì 21 maggio 2020, 15:44:51 CEST, TFlorian <notifications@github.com> ha scritto:  
 
 


If you use windows:

**Comment by 64Florian** [comment]
> do you want this folder?

- Yes please
- and your save of your project (add your stl in cura select your settings and click on File -> save )
- Are you under windows, Mac or linux ? 
- In you 1st message you sayed "only on the supports" and for your object it's OK ?? -> can you share a photo of your print  ?
- When you change your extruder have you keep the same "model" (titan / direct)  -> please tack a photo of the original and the new one

**Comment by battistipaolo** [comment]
hi,I have problem to tranfer zip file widh you.

in this save I removed from the supports, "the connection of support lines".
the system is windows10.
in the photo on the left, previous print with the same settings, on the right the output part with the redundant supports.
compared to this print in the save (gcode) I removed from the supports, "the connection of support lines".
I had tried a print but had to stop it for the same reason.
as you can see from the photo the piece that does not need the supports has been worked well.
the other photo is the design printed with kisslicer64.
paolo battisti

**Comment by battistipaolo** [comment]
hi,
I have problem to tranfer zip file widh you.

in this save I removed from the supports, "the connection of support lines".
the system is windows10.
in the photo on the left, previous print with the same settings, on the right the output part with the redundant supports.
compared to this print in the save (gcode) I removed from the supports, "the connection of support lines".
I had tried a print but had to stop it for the same reason.
as you can see from the photo the piece that does not need the supports has been worked well.

the other photo is the design printed with kisslicer64.

paolo battisti

[cura.zip](https://github.com/Ultimaker/Cura/files/4663892/cura.zip)

**Comment by battistipaolo** [comment]
Hi,
sorry
here are the configuration files
[4.6.zip](https://github.com/Ultimaker/Cura/files/4663927/4.6.zip)

**Comment by 64Florian** [comment]
Thank you for those files 
For "cura with supports" (in the cura.zip) as I can see you have enable support AND tree support ...
I'm totally lost between the title of this topic and your results but we will try to back to basics ...
1) could you save your workspace (it's a 3mf file extension) with your model and share (as Zip file) in this ticket please ? 
![Save 3mf File](https://user-images.githubusercontent.com/34726380/82606386-46f4f300-9bb7-11ea-87d7-6de249672fed.png)

2)  In my point of view, you printer is not properly tuned (in the printer side not in software like cura) As I can read in your GCode, your printer is a anycubic_i3_mega Could you try to measure if your extruder is properly set in you printer ? 
You have to make a mark with a pen on your filament and measure from the extruder the size of filament you have marked
![20180811115237](https://user-images.githubusercontent.com/34726380/82608879-bec51c80-9bbb-11ea-9aa6-2b64fb7f444e.jpg)
**Unplug** the output of you extruder (_The extruder must be able to run the filament as fast as he like_)
lunch this sample GCode: 
[ExtruderTest.gcode.zip](https://github.com/Ultimaker/Cura/files/4665023/ExtruderTest.gcode.zip)
The aim of this gcode is to ask to extract 100mm (10 cm) of plastique, This will allow us to know if motor extruder ration is properly defined in your printer ...
(The code made a Home, move up the nozzle 3mm, then extrude 10cm of plastic - cold bead and cold nozzle )
If all work as expected in your printer, you must have 10cm of plastic in one side and you mark move up by 10cm:
![20180811115815](https://user-images.githubusercontent.com/34726380/82610609-fda8a180-9bbe-11ea-8ac5-9e0616e335b1.jpg)

(images from https://www.rctruckstips.com/calibrer-son-extrudeur-pour-ameliorer-la-qualite-dimpression.html )

**Comment by battistipaolo** [comment]
good morning,
I will send you a file. zip with all your requests.
the file you sent (ExtruderTest.gcode) from me doesn't work, I tried to modify it, but it doesn't work.
in every way I have tried with repetier 100mm. = 96.61.
with the modification "M92 X81.6102 Y81.6102 Z400.1320 E95.5738; SET QUOTE XYZ E" that I inserted in my start gcode 100mm. = 100mm. of extrusion spill, so I would say that the extruder quota is right.
regards
paolo battisti
[E.zip](https://github.com/Ultimaker/Cura/files/4667635/E.zip)

**Comment by 64Florian** [comment]
>  I would say that the extruder quota is right

This is a really good news and a good start 

I discovered the "strange" settings ho disturb me so mutch .. it's "support wall line count"

![Capture d’écran 2020-05-22 à 13 17 34](https://user-images.githubusercontent.com/34726380/82663813-acd19100-9c30-11ea-84d9-557a0c2332fa.png)
![Capture d’écran 2020-05-22 à 13 17 59](https://user-images.githubusercontent.com/34726380/82663814-ad6a2780-9c30-11ea-9ea4-a508e6d4ebc9.png)
![Capture d’écran 2020-05-22 à 13 18 15](https://user-images.githubusercontent.com/34726380/82663811-aba06400-9c30-11ea-9c05-a4d5cd6f1bcb.png)
![Capture d’écran 2020-05-22 à 13 18 47](https://user-images.githubusercontent.com/34726380/82663807-a80cdd00-9c30-11ea-95bd-bc39f0e43603.png)

Now back to your bug: over-extrude 
In fact when the support wall is disable, Cura seems to make "double" support:
Stat (in blue) the wall is done 1st time 
![Capture d’écran 2020-05-22 à 13 35 35](https://user-images.githubusercontent.com/34726380/82664139-59137780-9c31-11ea-84b4-210f79632203.png)
And after the wall is re made a second time 
![Capture d’écran 2020-05-22 à 13 36 18](https://user-images.githubusercontent.com/34726380/82664142-5a44a480-9c31-11ea-951e-517ddd4345d8.png)

So if you want, enable wall end disable tree mode, you'll have this:
<img width="556" alt="Capture d’écran 2020-05-22 à 13 41 09" src="https://user-images.githubusercontent.com/34726380/82664395-eeaf0700-9c31-11ea-94af-67793284a8e2.png">
If I tune a bit to have a "better" support: 
<img width="1057" alt="Capture d’écran 2020-05-22 à 13 42 20" src="https://user-images.githubusercontent.com/34726380/82664484-1a31f180-9c32-11ea-98af-6b23d5d8727c.png">

You may need to change support orientation (in fact I didn't find the settings ... but you can rotate your object 45°) to be stronger / cleaner:
<img width="1065" alt="Capture d’écran 2020-05-22 à 13 46 32" src="https://user-images.githubusercontent.com/34726380/82664739-ab08cd00-9c32-11ea-9cec-22d482dbe730.png">

**Comment by battistipaolo** [comment]
HI Florian,
question: when is it possible to eliminate the double extrusion in the media settings that you have gradually modified?
or: what must be eliminated in the support settings to avoid double extrusion?
because at the moment I have many ideas in my head but all confused
regards
paolo battisti

**Comment by battistipaolo** [comment]
Hi Florian,
therefore: do use the supports without trees, or use the trees without supports, never both together?
Do I'm on the right track?
regards
paolo

**Comment by 64Florian** [comment]
Hello @Ghostkeeper, In my point of view, tree mode shouldn't be affected by a support wall count defined to 0 ... So it may be interesting to unplug this stetting from "support" and create a new entry "Tree wall line count"

**Comment by 64Florian** [comment]
> therefore: do use the supports without trees, or use the trees without supports, never both together?

not really... 
Using both it's really too much ! (in my point of view)
-> in fact, you "can" use both but you Mustn't use "Support wall count" to 0 with tree mode enable

so, as general idea:
- OK for tree + support With "support wall count" = 1
- OK for tree With "support wall count" = 1
- OK for support With "support wall count" = 1
- OK for support With "support wall count" = 0
- **bad** for **tree** With **"support wall count" = 0**
- **bad** for **tree** + support With **"support wall count" = 0**

In your case you need tree to have a good "window" 
![Capture d’écran 2020-05-22 à 15 57 18](https://user-images.githubusercontent.com/34726380/82675671-6fc3c980-9c45-11ea-9aba-c0506cb9fbe8.png)
(Only tree with "support wall count" = 1)

or you have to enable "everyway" support feature like this:
![Capture d’écran 2020-05-22 à 15 59 00](https://user-images.githubusercontent.com/34726380/82675693-78b49b00-9c45-11ea-913e-318878af5b05.png)
(no tree required)

**Comment by battistipaolo** [comment]
Hi,
 ok let's say the base is this:
support wall count "= always at 1 to avoid double extrusion of the supports, and" strange "printing of the supports.
ok I thank you immensely for solving the fix.
perhaps some changes remain in the Cura program to prevent this anomalous situation from being set, but this will in any case be evaluated by the programmers.
I consider the problem closed
thanks a lot.
regards
paolo battisti

**Comment by 64Florian** [comment]
Just for information, I found my setting for support line orientation 
-> this is better as turning the product on the build plate (_event more when you need all the size of the build plate to make you object_)
<img width="436" alt="Capture d’écran 2020-05-24 à 01 43 32" src="https://user-images.githubusercontent.com/34726380/82742563-506d8f00-9d60-11ea-95ba-b8b69f845596.png">

**Comment by battistipaolo** [comment]
Hi Florian,
sorry, if I understand correctly, 
which setting do I have to change to have the supports at 45 degrees instead of 90 degrees,
 instead of rotating the whole object?
regards
paolo battisti

**Comment by 64Florian** [comment]
You have to change "Support Infill Line Directions" from [] to [45] (for example)
Have a look:
This time I didn't rotate the piece (as you can see the build plate grid stay in the same orientation)
<img width="1020" alt="Capture d’écran 2020-05-24 à 13 26 13" src="https://user-images.githubusercontent.com/34726380/82752989-6b70eb00-9dc2-11ea-849f-5c77c68aff09.png">
<img width="1015" alt="Capture d’écran 2020-05-24 à 13 26 31" src="https://user-images.githubusercontent.com/34726380/82752991-6ca21800-9dc2-11ea-9456-0eb40558883b.png">

**Comment by 64Florian** [comment]
hem, just as tip (if you want a grate piece and didn't care about the orientation of the bottom) 
![Capture d’écran 2020-05-24 à 13 35 51](https://user-images.githubusercontent.com/34726380/82753169-f7374700-9dc3-11ea-816e-39614162f253.png)
![Capture d’écran 2020-05-24 à 13 37 21](https://user-images.githubusercontent.com/34726380/82753250-8cd2d680-9dc4-11ea-8f6e-b1f1e94abb36.png)
![Capture d’écran 2020-05-24 à 13 37 53](https://user-images.githubusercontent.com/34726380/82753251-8d6b6d00-9dc4-11ea-870e-e49745672885.png)

**Comment by 64Florian** [comment]
I have edited my last post: when you define an angle, if you want to change on each layers you have to define it (_initially I write to put only 0° angle_)

[Floiran_AI3M_portamicrotitan.3mf.zip](https://github.com/Ultimaker/Cura/files/4673410/Floiran_AI3M_portamicrotitan.3mf.zip)

Note: for an "Ultimate resistance part" you can use 0°, 90°, 45°,135° (double cross matrix)
![Capture d’écran 2020-05-24 à 13 59 19](https://user-images.githubusercontent.com/34726380/82753584-fc49c580-9dc6-11ea-93cf-90b97ce897e9.png)
![Capture d’écran 2020-05-24 à 13 59 25](https://user-images.githubusercontent.com/34726380/82753585-fce25c00-9dc6-11ea-883a-f09d2c6b4583.png)
![Capture d’écran 2020-05-24 à 13 59 32](https://user-images.githubusercontent.com/34726380/82753586-fd7af280-9dc6-11ea-94a9-31ca2c44cec9.png)
![Capture d’écran 2020-05-24 à 13 59 41](https://user-images.githubusercontent.com/34726380/82753587-fd7af280-9dc6-11ea-9a1b-4d65257ef6c4.png)

**Comment by battistipaolo** [comment]
Hi Florian,
A)-ok perfect, I didn't put "45" inside the square brackets, so it didn't work. you must excuse me but I am a little old man and my brain is no longer like when I was young.
B)-you are a mine of ideas, 
to improve the workmanship of the printer, you are truly the best. 
Thanks again for all the advice you have given me.
regards
paolo battisti

**Comment by 64Florian** [comment]
For the support,
> I didn't put "45" inside the square brackets

In fact you have 2 ways: 
- rotate your pice on the build place
- use the  "Support Infill Line Directions"

**Comment by battistipaolo** [comment]
Hi,
if I use "Support Infill Line Directions" 45 it doesn't work.
if I use "Support Infill Line Directions" [45] it works
the latter, it seems to me, is an excellent solution.

paolo

**Comment by Ghostkeeper** [comment]
> therefore: do use the supports without trees, or use the trees without supports, never both together?
> Do I'm on the right track?

Yeah, we plan to enforce that. It's one of the requirements for us to get Tree Support out of the experimental phase. We'd have one button to enable/disable support and a drop-down to select whether you want the straight support or tree. This is actually the same reason as this one:

> Hello @Ghostkeeper, In my point of view, tree mode shouldn't be affected by a support wall count defined to 0 ... So it may be interesting to unplug this stetting from "support" and create a new entry "Tree wall line count"

For that same reason, we removed the Tree Support Wall Line Count setting from Cura in 4.5. Once the two are together, it'll be a bit more logical. For us now, this merger of the two settings was an easy and logical way for us to fix Support Meshes when tree support was enabled.

> Hi,
if I use "Support Infill Line Directions" 45 it doesn't work.
if I use "Support Infill Line Directions" [45] it works
the latter, it seems to me, is an excellent solution.

Yeah, we're not enforcing the correct syntax with that setting I'm afraid. It's a bit hairy to do that per setting for us, in our code.

**Comment by battistipaolo** [comment]
Hi,
A) - I have already modified my default programs, I use the supports or the tree never together.
B) - I have defined at 1 "support wall count" and I have removed the check so it no longer appears selectable, so I avoid the error.
C) - the square bracket is not a big problem, just know that it is mandatory for multiple selections.

otherwise everything is ok, the print quality has improved a lot.
I thank you again for your help.
regards
paolo battisti
